src/main/java/org/openhab/binding/zwave/internal/protocol/ZWaveNode.java

```diff
@@ -1475,27 +1477,41 @@ public void run() {
                 logger.trace("NODE {}: WakeupTimerTask Already asleep", getNodeId());
                 return;
             }
+            count = count + 1;
+            logger.debug("NODE {}: WakeupTimerTask {} Messages waiting, state {} count {}", getNodeId(),
+                    controller.getSendQueueLength(getNodeId()), getNodeInitStage(), count);
 
-            logger.debug("NODE {}: WakeupTimerTask {} Messages waiting, state {}", getNodeId(),
-                    controller.getSendQueueLength(getNodeId()), getNodeInitStage());
-            if (triggered == false) {
-                logger.trace("NODE {}: WakeupTimerTask First iteration", getNodeId());
+            if (isInitializationComplete() == true  && controller.getSendQueueLength(getNodeId()) == 0) {
+                triggered = true;
+            }     
+            if (count == awakemax || count == 40) {
```

Comment:
Why count = 40?
Constants should be defined in the top of the class.

Reply:
Looking at this some more, I don't understand why you have the `|| count == 40` part at all. This seems to be another catchall in case `awakemax` is set to some random number, but that's not possible as this configuration is already limited to 20 seconds.

I would suggest to remove this check as it effectively conflicts with the configuration setting, and it means if we ever were to change the configuration, it wouldn't work due to this check.

Reply:
Done - I took this out.  It was a redundant backstop, but the awakeMax should be enough.
```
            if (count == awakeMax) {
                triggered = true;
            }
```